OpenGL-Core/src/GLCore/Events/Event.h

```diff
@@ -29,7 +29,7 @@ namespace GLCore {
 		EventCategoryMouseButton    = BIT(4)
 	};
 
-#define EVENT_CLASS_TYPE(type) static EventType GetStaticType() { return EventType::##type; }\
+#define EVENT_CLASS_TYPE(type) static EventType GetStaticType() { return EventType::type; }\
 								virtual EventType GetEventType() const override { return GetStaticType(); }\
```

Comment:
I believe that Yan is using the token-pasting operator (##) to concat EventType:: with whatever is the name pasted as type argument, so I wouldn't remove the operator here.

Reply:
The `##` operator is proprietary to Microsoft's compiler!

Reply:
Hey, I am not so sure about it tho as I have used it with gcc and clang professionally under various architecture implementations of the compiler standard. Here's an example code that makes use of the `##` operator compiled using `gcc-7`: https://godbolt.org/z/TEGszs

Also, the cppreference preprocessor docs don't seem to say anything about compiler specifics: https://en.cppreference.com/w/cpp/preprocessor/replace

And I have personally compiled the code under clang and gcc on both Linux (gcc and clang) and MacOS (clang) as is.

Reply:
I tested this operator on Godbolt during the summer and found that `##` works on the most recent few major versions of GCC. The above is `::##` which is different.

Reply:
Ah sweet, well not cool but ... 😆 I am wondering what could've made this compile fine for me, as far as I remember tho.
I guess there's nothing more that can be done for the above compiler feature support war. I've been misled :(